src/brevitas/quant/solver/common.py

```diff
@@ -289,19 +316,5 @@ def device(tracked_parameter_list):
 class SolveScaleSignedness(ExtendedInjector):
 
     @value
-    def is_scale_unsigned(scaling_stats_impl=None, scaling_init=None):
-        if scaling_init is not None:
-            if not isinstance(scaling_init, torch.Tensor):
-                scaling_init = torch.tensor(scaling_init)
-
-            # Check if any of the init values is negative
-            if scaling_init.shape == ():
-                is_scale_negative = scaling_init < 0
-            else:
-                is_scale_negative = any(scaling_init.flatten() < 0)
-            return not is_scale_negative
-        elif hasattr(scaling_stats_impl, 'is_scale_unsigned'):
-            return scaling_stats_impl.is_scale_unsigned
-        else:
-            # If it is not possible to infer the scale of the sign, we assume it is unsigned
-            return True
+    def is_scale_unsigned(restrict_scaling_type=None):
+        return restrict_scaling_type in [RestrictValueType.FP, RestrictValueType.POWER_OF_TWO]
```

Comment:
This is resolved by looking at the register dict

Reply:
We need to register statistics with a decorator and put them in a dict that tells us if they are signed or unsigned.
We should use that dict for this check.
If the stats is not registered, this fails